demos/dev_mode_key_provisioning/src/aws_dev_mode_key_provisioning.c

```diff
@@ -171,16 +171,34 @@ static CK_RV prvProvisionPrivateECKey( CK_SESSION_HANDLE xSession,
 
     if( xResult == CKR_OK )
     {
-        CK_ATTRIBUTE xPrivateKeyTemplate[] =
-        {
-            { CKA_CLASS,     &xPrivateKeyClass, sizeof( CK_OBJECT_CLASS )                        },
-            { CKA_KEY_TYPE,  &xPrivateKeyType,  sizeof( CK_KEY_TYPE )                            },
-            { CKA_LABEL,     pucLabel,          ( CK_ULONG ) strlen( ( const char * ) pucLabel ) },
-            { CKA_TOKEN,     &xTrue,            sizeof( CK_BBOOL )                               },
-            { CKA_SIGN,      &xTrue,            sizeof( CK_BBOOL )                               },
-            { CKA_EC_PARAMS, pxEcParams,        EC_PARAMS_LENGTH                                 },
-            { CKA_VALUE,     pxD,               EC_D_LENGTH                                      }
-        };
+        CK_ATTRIBUTE xPrivateKeyTemplate[ 7 ] = { 0 };
```

Comment:
Expanding the assignment is adding a lot of lines. The aggregate initialization provides good readability. 
To retain the aggregate initialization, an alternative approach could be like:
```
CK_ATTRIBUTE xPrivateKeyTemplate[] =
{
    { CKA_CLASS,     NULL /* &xPrivateKeyClass*/, sizeof( CK_OBJECT_CLASS )                        },
    { CKA_KEY_TYPE,  NULL /* &xPrivateKeyType*/,  sizeof( CK_KEY_TYPE )                            },
    { CKA_LABEL,     NULL /* pucLabel*/,          ( CK_ULONG ) strlen( ( const char * ) pucLabel ) },
    { CKA_TOKEN,     NULL /* &xTrue*/,            sizeof( CK_BBOOL )                               },
    { CKA_SIGN,      NULL /* &xTrue*/,            sizeof( CK_BBOOL )                               },
    { CKA_EC_PARAMS, NULL /* pxEcParams*/,        EC_PARAMS_LENGTH                                 },
    { CKA_VALUE,     NULL /* pxD*/,               EC_D_LENGTH                                      }
};

xPrivateKeyTemplate[ 0 ].pValue = &xPrivateKeyClass;
xPrivateKeyTemplate[ 1 ].pValue = &xPrivateKeyType;
xPrivateKeyTemplate[ 2 ].pValue = pucLabel;
xPrivateKeyTemplate[ 3 ].pValue = &xTrue;
xPrivateKeyTemplate[ 4 ].pValue = &xTrue;
xPrivateKeyTemplate[ 5 ].pValue = pxEcParams;
xPrivateKeyTemplate[ 6 ].pValue = pxD;
```